src/overlays/actors/ovl_En_Rd/z_en_rd.h

```diff
@@ -8,29 +8,75 @@ struct EnRd;
 
 typedef void (*EnRdActionFunc)(struct EnRd*, GlobalContext*);
 
+#define EN_RD_GET_FLAGS(thisx) (((thisx)->params & 0xFF00) >> 8)
+
+typedef enum GibdoLimbs {
+    /*  0 */ REDEAD_GIBDO_LIMB_NONE,
+    /*  1 */ REDEAD_GIBDO_LIMB_ROOT,
+    /*  2 */ REDEAD_GIBDO_LIMB_LEFT_LEG_ROOT,
+    /*  3 */ REDEAD_GIBDO_LIMB_LEFT_THIGH,
+    /*  4 */ REDEAD_GIBDO_LIMB_LEFT_SHIN,
+    /*  5 */ REDEAD_GIBDO_LIMB_LEFT_FOOT_ROOT,
+    /*  6 */ REDEAD_GIBDO_LIMB_LEFT_FOOT,
+    /*  7 */ REDEAD_GIBDO_LIMB_RIGHT_LEG_ROOT,
+    /*  8 */ REDEAD_GIBDO_LIMB_RIGHT_THIGH,
+    /*  9 */ REDEAD_GIBDO_LIMB_RIGHT_SHIN,
+    /* 10 */ REDEAD_GIBDO_LIMB_RIGHT_FOOT_ROOT,
+    /* 11 */ REDEAD_GIBDO_LIMB_RIGHT_FOOT,
+    /* 12 */ REDEAD_GIBDO_LIMB_UPPER_BODY_ROOT,
+    /* 13 */ REDEAD_GIBDO_LIMB_TORSO_ROOT,
+    /* 14 */ REDEAD_GIBDO_LIMB_TORSO,
+    /* 15 */ REDEAD_GIBDO_LIMB_LEFT_ARM_ROOT,
+    /* 16 */ REDEAD_GIBDO_LIMB_LEFT_UPPER_ARM,
+    /* 17 */ REDEAD_GIBDO_LIMB_LEFT_FOREARM,
+    /* 18 */ REDEAD_GIBDO_LIMB_LEFT_HAND,
+    /* 19 */ REDEAD_GIBDO_LIMB_RIGHT_ARM_ROOT,
+    /* 20 */ REDEAD_GIBDO_LIMB_RIGHT_UPPER_ARM,
+    /* 21 */ REDEAD_GIBDO_LIMB_RIGHT_FOREARM,
+    /* 22 */ REDEAD_GIBDO_LIMB_RIGHT_HAND,
+    /* 23 */ REDEAD_GIBDO_LIMB_HEAD_ROOT,
+    /* 24 */ REDEAD_GIBDO_LIMB_HEAD,
+    /* 25 */ REDEAD_GIBDO_LIMB_PELVIS,
+    /* 26 */ REDEAD_GIBDO_LIMB_MAX
+} RedeadGibdoLimbs;
+
+typedef enum {
+    /* -3 */ EN_RD_TYPE_GIBDO_RISING_OUT_OF_COFFIN = -3,
+    /* -2 */ EN_RD_TYPE_GIBDO = -2,
+    /* -1 */ EN_RD_TYPE_DOES_NOT_MOURN = -1,
+    /*  0 */ EN_RD_TYPE_DOES_NOT_MOURN_IF_WALKING = 0,
+    /*  1 */ EN_RD_TYPE_REGULAR,
+    /*  2 */ EN_RD_TYPE_CRYING,
+    /*  3 */ EN_RD_TYPE_INVISIBLE
+} EnRdType;
+
 typedef struct EnRd {
     /* 0x0000 */ Actor actor;
     /* 0x014C */ Vec3s firePos[10];
     /* 0x0188 */ SkelAnime skelAnime;
-    /* 0x01CC */ Vec3s jointTable[26];
-    /* 0x0268 */ Vec3s morphTable[26];
-    /* 0x0304 */ u8 unk_304;
-    /* 0x0305 */ u8 unk_305;
-    /* 0x0306 */ u8 unk_306;
-    /* 0x0307 */ u8 unk_307;
+    /* 0x01CC */ Vec3s jointTable[REDEAD_GIBDO_LIMB_MAX];
+    /* 0x0268 */ Vec3s morphTable[REDEAD_GIBDO_LIMB_MAX];
+    /* 0x0304 */ u8 grabState;
+    /* 0x0305 */ u8 isMourning;
+    /* 0x0306 */ u8 playerStunWaitTimer; // Cannot stun the player if this is non-zero
+    /* 0x0307 */ u8 grabWaitTimer; // Cannot grab the player if this is non-zero
     /* 0x0308 */ EnRdActionFunc actionFunc;
-    /* 0x030C */ s16 unk_30C;
-    /* 0x030E */ s16 unk_30E;
-    /* 0x0310 */ s16 unk_310;
-    /* 0x0312 */ s16 unk_312;
-    /* 0x0314 */ s16 unk_314;
-    /* 0x0316 */ s16 unk_316;
-    /* 0x0318 */ u8 unk_318;
-    /* 0x0319 */ u8 unk_319;
-    /* 0x031A */ u8 unk_31A;
-    /* 0x031B */ u8 unk_31B;
-    /* 0x031C */ u8 unk_31C;
-    /* 0x031D */ u8 unk_31D;
+    /* 0x030C */ union {
+                    s16 animationJudder;
+                    s16 death;
+                    s16 coffinRiseForwardAccel;
+                } timer;
```

Comment:
we dont generally use unions soley for the purpose of giving a variable different names. At least its something we've actively avoided in the past, but iirc @Roman971 feels more strongly about that

Reply:
Yeah I'm personally not a fan of using unions for naming purposes since it could be misleading very quickly and lead to strange bugs in modding for instance. It's not as bad if there is a name for the union itself (ie. the `timer` part here), but it can still be confused with a struct named `timer` for example, so it's not ideal.

Here I would just go for a generic `timer` variable, but a union is probably okay if we consider the specific names important enough

Reply:
Other actors use defines for this sort of situation (EnTite and BossSst come to mind, there's probably a few more).

Reply:
in this specific case I would probably just say a variable named `timer` is best, its what we do in many other actors. generic timers are expected to be action dependent id say

Reply:
In this specific case, I'll relent and use `timer`. However, it's very, very clear that this approach will not scale with every actor; take a look at Obj_Tokeidai in MM and try to figure out how you would document all that without a union or a define.